js/ui/payg.js

```diff
@@ -612,6 +615,271 @@ var ApplyCodeNotification = GObject.registerClass({
     }
 });
 
+/* ModalDialog -> PaygAddCreditDialog
+ *
+ * A modal dialog for entering a code while the computer is unlocked.
+ *
+ * ModalDialog.dialogLayout._dialog
+ * ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+ * ┃                                                                           ┃
+ * ┃ ModalDialog.contentLayout                                                 ┃
+ * ┃ ┌───────────────────────────────────────────────────────────────────────┐ ┃
+ * ┃ │PaygAddCreditDialog._promptLayout (Dialog.MessageDialogContent)        │ ┃
+ * ┃ │┌─────────────────────────────────────────────────────────────────────┐│ ┃
+ * ┃ ││                       Enter your unlock code                        ││ ┃
+ * ┃ ││                                                                     ││ ┃
+ * ┃ ││          Text label that either prompts the user for action         ││ ┃
+ * ┃ ││                          or shows results.                          ││ ┃
+ * ┃ │└─────────────────────────────────────────────────────────────────────┘│ ┃
+ * ┃ │PaygAddCreditDialog._codeEntryLayout                                   │ ┃
+ * ┃ │┌─────────────────────────────────────────────────────────────────────┐│ ┃
+ * ┃ ││     ┌─────────────────────────────────────────────────────┐         ││ ┃
+ * ┃ ││  *  │                                                     │  #      ││ ┃
+ * ┃ ││     └─────────────────────────────────────────────────────┘         ││ ┃
+ * ┃ │└─────────────────────────────────────────────────────────────────────┘│ ┃
+ * ┃ └───────────────────────────────────────────────────────────────────────┘ ┃
+ * ┃ ModalDialog.buttonLayout                                                  ┃
+ * ┃ ┌───────────────────────────────────┬───────────────────────────────────┐ ┃
+ * ┃ │                                   │                                   │ ┃
+ * ┃ │              Cancel               │            Add Credit             │ ┃
+ * ┃ │                                   │                                   │ ┃
+ * ┃ └───────────────────────────────────┴───────────────────────────────────┘ ┃
+ * ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
+ */
+var PaygAddCreditDialog = GObject.registerClass(
+class PaygAddCreditDialog extends ModalDialog.ModalDialog {
+    _init() {
+        /* We want to be able to open the dialog multiple times per session
+         * without making the caller instatiate a new object before every call,
+         * so we need to disable destroyOnClose */
+        super._init({ styleClass: 'payg-add-credit-dialog',
+                      destroyOnClose: false });
+        super.connect('closed', this._onClosed.bind(this));
+
+        this.verificationStatus = UnlockStatus.NOT_VERIFYING;
```

Comment:
Is `verificationStatus` meant to be accessed by other classes? If not, please switch to `_verificationStatus`; if yes, please add a GObject property

Reply:
It is not, I'll make it private.